### PR TITLE
Add notes to the Trigger Dag UI form

### PR DESCRIPTION
**Pull request by shubhamraj-git**
Add notes to the Trigger Dag UI form

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx [hunk 1]
```diff
@@ -204,6 +204,21 @@ const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
...
+              <Text fontSize="md" mb={2} mt={6}>
+                Notes
+              </Text>
+              <Controller
+                control={control}
+                name="notes"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="Optional- Provide context for triggering this DAG run (e.g., 'Testing feature X', 'Backfilling data for 2024-01-01')"
```
Comment (bbovenzi) [inline airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: I don't think the placeholder is needed. Everything else is optional too.

Comment (shubhamraj-git) [reply airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: Rational behind was sometimes user get overwhelm by what to write and leaves it empty, so thought to give them idea, what do you think?

Comment (bbovenzi) [reply airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: I'd rather keep it simple for now.

Comment (jedcunningham) [reply airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: Same, keep it simple. Maybe keep "Optional" though.

Comment (shubhamraj-git) [reply airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: Makes sense, I ll make the changes.

File: airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx [hunk 2]
```diff
@@ -204,6 +204,17 @@ const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
                   </Box>
                 )}
               />
+
+              <Text fontSize="md" mb={2} mt={6}>
+                Notes
```
Comment (jscheffl) [inline airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: Can we make clear that this are not just "Any Notes" but "Dag Run Notes"?

(If I correctly assume from view of legacy UI this refers to the notes as below?)
image

```suggestion
                Dag Run Notes
```

**Review by jscheffl — APPROVED** [review]

File: airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx (hunk 1 above)
Comment (jscheffl) [reply airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx]: Haha, funny compared to @jedcunningham I actually had exactly the reverse opinion... I'd have removed the "Optional" and left the description what this is for. But not critical. Headline is adjusted, so for me this also has sufficient context now.

**Pull request merged**